### PR TITLE
refactor(generator): remove range type aliases

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5820)

### CODE REVIEW

File: generator/integration_tests/golden/golden_idempotency_policy_test.cc [hunk 1]
```diff
@@ -119,8 +119,8 @@ TEST_F(GoldenIdempotencyPolicyTest, ListBackupOperations) {
   EXPECT_EQ(policy_->ListBackupOperations(request), Idempotency::kIdempotent);
 }
 
-}  // namespace
-}  // namespace golden_internal
-}  // namespace GOOGLE_CLOUD_CPP_NS
-}  // namespace cloud
-}  // namespace google
+} // namespace
```
Comment (devjgm) [inline generator/integration_tests/golden/golden_idempotency_policy_test.cc]: I think two spaces before the EOL comment like it was before is preferred in the style guide.

File: generator/integration_tests/golden/golden_idempotency_policy_test.cc [hunk 2]
```diff
@@ -26,7 +26,7 @@ using google::cloud::internal::Idempotency;
 namespace gtab = ::google::test::admin::database::v1;
 
 class GoldenIdempotencyPolicyTest : public ::testing::Test {
- protected:
+protected:
```
Comment (devjgm) [inline generator/integration_tests/golden/golden_idempotency_policy_test.cc]: 1 space, as before, was style compliant.

File: generator/integration_tests/golden/golden_logging_decorator_test.cc [hunk 3]
```diff
@@ -31,136 +31,135 @@ using ::testing::Return;
...
-       ::google::test::admin::database::v1::GetDatabaseRequest const& request),
+       ::google::test::admin::database::v1::GetDatabaseRequest const &request),
       (override));
 
   MOCK_METHOD(
       StatusOr<::google::longrunning::Operation>, UpdateDatabaseDdl,
       (grpc::ClientContext & context,
-       ::google::test::admin::database::v1::UpdateDatabaseDdlRequest const&
-           request),
+       ::google::test::admin::database::v1::UpdateDatabaseDdlRequest const
```
Comment (devjgm) [inline generator/integration_tests/golden/golden_logging_decorator_test.cc]: Some formatting looks strange w/ the placement of the ampersand. Also on the previous line where there are spaces on both sides of the ampersand.

File: generator/integration_tests/golden/database_admin_connection.gcpcxx.pb.cc [hunk 4]
```diff
@@ -434,7 +440,8 @@ class DatabaseAdminConnectionImpl : public DatabaseAdminConnection {
         backoff_policy_prototype_->clone());
     auto idempotency = idempotency_policy_->ListBackups(request);
     char const* function_name = __func__;
-    return google::cloud::internal::MakePaginationRange<ListBackupsRange>(
+    return google::cloud::internal::MakePaginationRange<StreamRange<
```
Comment (devjgm) [inline generator/integration_tests/golden/database_admin_connection.gcpcxx.pb.cc:443]: Can we change the line break here to: 

```cc
    return google::cloud::internal::MakePaginationRange<
        StreamRange<::google::test::admin::database::v1::Backup>>(
```

Which I think is what clang-format would give us.

File: generator/internal/connection_generator.cc [hunk 5]
```diff
@@ -51,10 +51,11 @@ Status ConnectionGenerator::GenerateHeader() {
        vars("retry_traits_header_path"), "google/cloud/backoff_policy.h",
        "google/cloud/connection_options.h",
        HasLongrunningMethod() ? "google/cloud/future.h" : "",
-       HasPaginatedMethod() ? "google/cloud/internal/pagination_range.h" : "",
        HasLongrunningMethod() ? "google/cloud/polling_policy.h" : "",
        "google/cloud/status_or.h",
-       HasStreamingReadMethod() ? "google/cloud/stream_range.h" : "",
+       HasStreamingReadMethod() || HasPaginatedMethod()
+           ? "google/cloud/stream_range.h"
```
Comment (devjgm) [inline generator/internal/connection_generator.cc:57]: Note, we'll still need to include `google/cloud/internal/pagination_range.h` in any file that calls the `MakePaginationRange()` function.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5820)* status: 0 of 15 files reviewed, 2 unresolved discussions (waiting on @devjgm)

___
*[generator/integration_tests/golden/database_admin_connection.gcpcxx.pb.cc, line 443 at r3](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5820#-MSoD8QZeU99hnpT_MSV-r3-443:-MSoOIcA-g2Nr_CBD7sS:b66i71v) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/1e1442744312696ab6a93ec85684a404baa1c7eb/generator/integration_tests/golden/database_admin_connection.gcpcxx.pb.cc#L443)):*
<details><summary><i>Previously, devjgm (Greg Miller) wrote…</i></summary><blockquote>

Can we change the line break here to: 

```cc
    return google::cloud::internal::MakePaginationRange<
        StreamRange<::google::test::admin::database::v1::Backup>>(
```

Which I think is what clang-format would give us.
</blockquote></details>

Not going to worry about formatting in generated files at this time.
___
*[generator/internal/connection_generator.cc, line 57 at r3](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5820#-MSoD8QZeU99hnpT_MSl-r3-57:-MSoO9JfD1-iuY4XOzqf:bwxef4b) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/1e1442744312696ab6a93ec85684a404baa1c7eb/generator/internal/connection_generator.cc#L57)):*
<details><summary><i>Previously, devjgm (Greg Miller) wrote…</i></summary><blockquote>

Note, we'll still need to include `google/cloud/internal/pagination_range.h` in any file that calls the `MakePaginationRange()` function.
</blockquote></details>

Correct. The generated .cc file includes pagination_range.h the generated .h file does not need to.

File: generator/internal/connection_generator.cc [hunk 6]
```diff
@@ -289,9 +279,10 @@ Status ConnectionGenerator::GenerateCc() {
          MethodPattern(
              {
                  // clang-format off
-   {"$method_name$Range $connection_class_name$::$method_name$(\n"
+   {"StreamRange<$range_output_type$> $connection_class_name$::$method_name$(\n"
     "    $request_type$ request) {\n"
-    "  return google::cloud::internal::MakePaginationRange<$method_name$Range>(\n"
+    "  return google::cloud::internal::MakePaginationRange<StreamRange<\n"
```
Comment (devjgm) [inline generator/internal/connection_generator.cc:284]: Looks like it should be easy to fix this formatting here and one line 447.

**Review by devjgm — APPROVED** [review]

**Pull request merged**